private-templates-service/server/src/db/providers/StorageProvider.ts

```diff
@@ -0,0 +1,175 @@
+// Base class for all the StorageProviders.
+import logger from '../../util/logger'
+module Interface {
+
+    export interface IUser  {
+        _id?: string,
+        team: string[],
+        org: string[],
+        email: string
+    }
+
+    export interface ITemplateInstance {
+        json: string,
+        version: string
+    }
+
+    export interface ITemplate {
+        _id?: string,
+        instances: ITemplateInstance[],
+        tags: string[],
+        owner: string,
+        createdAt: Date,
+        updatedAt: Date,
+        isPublished: boolean
+    }
+}
+
+module Queries {
+
+    export interface TemplateQuery {
+        templateID?: string,
+        version?: string,
+        tags?: string[],
+        owner?: string,
+        isPublished? : boolean,
+        //sort based on date
+        // permissions? : [enum]
+    }
+
+    export interface UserQuery {
+        userID? : string,
+        teamID? : string[],
+        orgID? : string[],
+        email?: string
+    }
+
+    // In case we need folders
```

Comment:
Just remove this and add it in later if you want, no point having commented code cluttering it up